src/kubectl.ts

```diff
@@ -227,6 +232,17 @@ async function asLines(context: Context, command: string): Promise<Errorable<str
     return { succeeded: false, error: [ shellResult.stderr ] };
 }
 
+async function fromLines(context: Context, command: string): Promise<Errorable<{ [key: string]: string }[]>> {
+    const shellResult = await invokeAsync(context, command);
+    if (shellResult.code === 0) {
+        let lines = shellResult.stdout.split('\n');
+        lines = lines.filter((l) => l.length > 0);
```

Comment:
@itowlson I missed checking for empty response here, which will break `parseLineOutput`.
Thinking about following solution or open to other solutions: 
```typescript
let parsedOutput = [];
        if (lines.length > 0) {
            parsedOutput = parseLineOutput(lines);
        }
        return { succeeded: true, result: parsedOutput};
```

Reply:
Good catch.  You're right, if there are no resources then there is no header line.  I'd actually consider putting the fix into `kubectlUtils.parseLineOutput` -  you do `const headers = lineOutput.shift()` to get the header line, and checking `if (!headers)` should solve the problem nicely (and play better with the typechecker once Project Strict Nulls gets that far) - I am pretty sure `shift` returns `undefined` if there are no array elements...?

Reply:
sounds good, testing this approach.